server/server/src/discovery.rs

```diff
@@ -0,0 +1,66 @@
+use std::{
+    fmt::Display,
+    net::{IpAddr, Ipv4Addr},
+};
+
+use local_ip_address::list_afinet_netifas;
+use service::settings::ServerSettings;
+
+#[derive(Clone)]
+pub enum Protocol {
+    Http,
+    Https,
+}
+
+impl Display for Protocol {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "{}",
+            match self {
+                Protocol::Http => "http",
+                Protocol::Https => "https",
+            }
+        )
+    }
+}
+
+#[derive(Clone)]
+pub struct ServerInfo {
+    protocol: Protocol,
+    port: u16,
+    ip: Ipv4Addr,
+}
+
+impl ServerInfo {
+    pub fn new(protocol: Protocol, settings: &ServerSettings) -> Self {
+        ServerInfo {
+            protocol,
+            port: settings.port,
+            ip: get_local_ip(),
+        }
+    }
+
+    pub fn as_url(&self) -> String {
+        format!("{}://{}:{}", self.protocol, self.ip, self.port)
+    }
+}
+
+fn get_local_ip() -> Ipv4Addr {
+    let network_interfaces = list_afinet_netifas().unwrap();
+
+    let v4_ips: Vec<Ipv4Addr> = network_interfaces
+        .into_iter()
+        .filter_map(|(_, ip)| match ip {
+            IpAddr::V4(ip) => Some(ip),
```

Comment:
do we need to support v6 at some point?

Reply:
This one is meant to be for local network, i think v4 should be good ?

Reply:
I guess so if we need ipv6 we can add it then